packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/InAppPurchasePlugin.java

```diff
@@ -103,4 +110,39 @@ private void teardownMethodChannel() {
   void setMethodCallHandler(MethodCallHandlerImpl methodCallHandler) {
     this.methodCallHandler = methodCallHandler;
   }
+
+  private static String getLibraryPackageName() {
+    String packageName = null;
+    Exception errorGettingLibraryPackageName = null;
+    Exception errorGettingApplicationID = null;
+    try {
+      Field libraryPackageName = BuildConfig.class.getField("LIBRARY_PACKAGE_NAME");
```

Comment:
This BuildConfig comes from com.android.billingclient.api.BillingClient, right?

Then it ends up with PBL's package name, right?

Reply:
I think this `BuildConfig` comes from io.flutter.plugins.inapppurchase if I'm not mistaken? The test sorta proved it. Unless I'm missing something?

Reply:
My bad, I mistook BillingClient as BuildConfig... 

I feel LIBRARY_PACKAGE_NAME is enough. In what case this field will be missing?

Reply:
I think some lower android studio version or lower gradle version only supports `APPLICATION_ID`? At least from my local machine, I didn't have `LIBRARY_PACKAGE_NAME`, I only have `APPLICATION_ID`

Reply:
`BuildConfig` comes from the [app package](https://developer.android.com/studio/build/gradle-tips#share-custom-fields-and-resource-values-with-your-app-code). It's generated by the Android Gradle tools.

@cyanglaz why is reflection required? Have you tried to build this for release? (R8 might obfuscate the class name)

Reply:
Make sense, APPLICATION_ID is deprecated since Android Studio 3.5.